sympy/core/power.py

```diff
@@ -1720,13 +1720,19 @@ def mul(d1, d2):
             return res
 
         try:
-            _, d = g.leadterm(x, logx=logx)
+            c, d = g.leadterm(x, logx=logx)
         except (ValueError, NotImplementedError):
             if limit(g/x**maxpow, x, 0) == 0:
                 # g has higher order zero
                 return f**e + e*f**e*g  # first term of binomial series
             else:
                 raise NotImplementedError()
+        if c.is_Float and d == S.Zero:
+            # Convert floats like 0.5 to exact SymPy numbers like S.Half, to
+            # prevent rounding errors which can induce wrong values of d leading
+            # to execution of an inappropriate code block (line 1741 - 1750)
```

Comment:
It's a bad idea to include line numbers like this. Those will become out of date as soon as anyone edits the code above here.

Reply:
Ahh sorry , yeah I missed that . I am going to make a pr today for implementing series for some functions where I'll address this too , thanks for pointing this out @oscarbenjamin !